java/wavefrontsender.md

```diff
@@ -0,0 +1,212 @@
+WavefrontProxyClient.Builder wfProxyClientBuilder = new WavefrontProxyClient.Builder(proxyHostName);
+
+// Set the proxy port to send metrics to. Default: 2878
+wfProxyClientBuilder.metricsPort(2878);
+
+// Set a proxy port to send histograms to.  Recommended: 2878
+wfProxyClientBuilder.distributionPort(2878);
+
+// Set a proxy port to send trace data to. Recommended: 30000
+wfProxyClientBuilder.tracingPort(30_000);
+
+// Optional: Set a custom socketFactory to override the default SocketFactory
+wfProxyClientBuilder.socketFactory(<SocketFactory>);
+
+// Optional: Set a nondefault interval (in seconds) for flushing data from the sender to the proxy. Default: 5 seconds
+wfProxyClientBuilder.flushIntervalSeconds(2);
+
+// Create the WavefrontProxyClient
+WavefrontSender wavefrontSender = wfProxyClientBuilder.build();
+ ```
+**Note:** When you [set up a Wavefront proxy](https://github.com/wavefrontHQ/java/tree/master/proxy#set-up-a-wavefront-proxy) on the specified proxy host, you specify the port it will listen to for each type of data to be sent. The `WavefrontProxyClient` must send data to the same ports that the Wavefront proxy listens to. Consequently, the port-related builder methods must specify the same port numbers as the corresponding proxy configuration properties: 
+
+| `WavefrontProxyClient` builder method | Corresponding property in `wavefront.conf` |
+| ----- | -------- |
+| `metricsPort()` | `pushListenerPorts=` |
+| `distributionPort()` | `histogramDistListenerPorts=` |
+| `tracingPort()` | `traceListenerPorts=` |
+ 
+
+## Option 2: Sending Data via Direct ingestion
+
+when sending data via direct ingestion, you need to create a `WavefrontDirectIngestionClient`, and build it with the information it needs to send data directly to Wavefront.
+
+### Step 1. Obtain Wavefront Access Information
+Gather the following access information:
+
+* Identify the URL of your Wavefront instance. This is the URL you connect to when you log in to Wavefront, typically something like `https://<domain>.wavefront.com`.
+* In Wavefront, verify that you have Direct Data Ingestion permission, and [obtain an API token](http://docs.wavefront.com/wavefront_api.html#generating-an-api-token).
+
+### Step 2. Initialize the WavefrontDirectIngestionClient
+You initialize a `WavefrontDirectIngestionClient` by building it with the access information you obtained in Step 1.
+
+You can optionally call builder methods to tune the following ingestion properties:
+
+* Max queue size - Internal buffer capacity of the `WavefrontSender`. Any data in excess of this size is dropped.
+* Flush interval - Interval for flushing data from the `WavefrontSender` directly to Wavefront.
+* Batch size - Amount of data to send to Wavefront in each flush interval.
+
+Together, the batch size and flush interval control the maximum theoretical throughput of the `WavefrontSender`. You should override the defaults _only_ to set higher values.
```

Comment:
... You should override the... => Override the... 
(We don't use "should". It implies moral obligation more than anything else, like, "you shouldn't eat too many sweets" ;->).